### PR TITLE
1.18/dev

### PR DESCRIPTION
**Pull request by desht**
Backport of my 1.19 changes.  Looks bigger but includes some earlier from Max/Mikey that I just needed to merge in just now.

### CODE REVIEW

**Comment by MaxNeedsSnacks** [comment]
This looks like something we should squash, doesn't it lol

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 plugins {
 	id "architectury-plugin" version "3.4-SNAPSHOT"
-	id "dev.architectury.loom" version "0.11.0-SNAPSHOT" apply false
+	id "dev.architectury.loom" version "0.12.0-SNAPSHOT" apply false
```
Comment (MaxNeedsSnacks) [inline build.gradle]: This can actually be bumped to `1.0-SNAPSHOT` now

File: common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java [hunk 2]
```diff
@@ -171,17 +210,86 @@ private static int listRanksOf(CommandSourceStack source, ServerPlayer player) {
...
+			source.sendSuccess(new TranslatableComponent("ftbranks.node_added", "condition", value, rank), false);
+		} catch (Exception e) {
+			throw new SimpleCommandExceptionType(new TextComponent(e.getMessage())).create();
+		}
+
+		return 1;
+	}
+
+	private static int showRank(CommandSourceStack source, Rank rank) {
+		source.sendSuccess(new TextComponent(Strings.repeat('=', 50)).withStyle(ChatFormatting.GREEN), false);
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: I originally missed this in the 1.19 PR, but this should probably not be using jopt.internal >.>

**Review by MaxNeedsSnacks — CHANGES_REQUESTED** [review]

File: common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java (hunk 2 above)
Comment (desht) [reply common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: Oops, yeah.  Meant to use `com.google.common.base.Strings` there

Comment (MaxNeedsSnacks) [reply common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: I think starting in Java... 9? 11? You can also just use String.repeat from Java's stdlib? :P

Comment (desht) [reply common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: I don't see a `String.repeat` method...

Oh wait, it's non-static.  `"=".repeat(50)` would do the trick.

Comment (MaxNeedsSnacks) [reply common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: It's non-static, so you'd have to use it on a String literal

Comment (desht) [reply common/src/main/java/dev/ftb/mods/ftbranks/FTBRanksCommands.java]: OK, now using `"=".repeat()`

**Review by MaxNeedsSnacks — APPROVED** [review]

**Pull request merged**